tensorflow_datasets/core/download/extractor.py

```diff
@@ -240,4 +276,6 @@ def iter_archive(
     raise ValueError(
         f'Cannot `iter_archive` over {path}. Invalid or unrecognised archive.'
     )
-  return _EXTRACT_METHODS[method](path)  # pytype: disable=bad-return-type
+  archive_file  = _get_archive_file(path, method)
+  kwargs = {'stream': True} if method.name.endswith('STREAM') else {}
+  return archive_file.get_iter(**kwargs)
```

Comment:
I'm not sure if this is the best way to implement this.
Another idea I had was for the `_TarFile` class to have a parameter `stream` and then the mapping could be changed to `_Tarfile(stream=True)` where required in `_EXTRACT_METHODS`

This would be much cleaner, but I'm not sure how to implement this properly. 
Other possibilities are 
- to have a function `_TarFile` which takes an argument `stream` and creates a class inside it bases on the arg and return that class. We should also memorize this so it returns the same class every time for the same parameter. 
- create two classes one for stream and one plain. But this would lead to a lot of code duplication